### PR TITLE
Modify kopia utilities to use new kopia cache flags

### PR DESCRIPTION
**Pull request by kale-amruta**
## Change Overview

Kopia has recently added a bunch of flags for cache settings while doing repository operations. This PR adds the options to set these flags in 
- Repository Create
- Repostory Connect
- Repository Server connect

Kopia Client related changes:
- Since some of the kanister functions use repository server connect command, there were changes required in these functions as well
- Kopia client changes in the function `ConnectToAPIServer`

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kopia/command/cache.go
Comment (pavannd1) [inline pkg/kopia/command/cache.go:9]: Let's add a comment here

File: pkg/kopia/command/repository.go
Comment (pavannd1) [inline pkg/kopia/command/repository.go:30]: Can we have two implicit variables?
I guess one's a pointer and other's a value so should work okay

File: go.mod
Comment (kale-amruta) [inline go.mod:33]: Ignore the changes in go.mod and go.sum, these are from Le's PR - https://github.com/kanisterio/kanister/pull/2416

File: go.sum
Comment (kale-amruta) [inline go.sum:21]: https://github.com/kanisterio/kanister/pull/2393/files#r1368642301

File: pkg/kopia/repository/client.go
Comment (ankitjain235) [inline pkg/kopia/repository/client.go:66]: I see `ContentCacheSizeBytes` and `MetadataCacheSizeBytes` in PR by @leuyentran : https://github.com/kanisterio/kanister/pull/2416/files#diff-d96603cf822203f32cb69e1031279e1c357193ab6a01bbcf59ce08caa3dc3f8dR66. Do we know which one of these two should be used?

**Review by ankitjain235 — CHANGES_REQUESTED** [review]
Just blocking merge of the PR until the confusion in flag names is cleared.

File: pkg/kopia/repository/client.go
Comment (kale-amruta) [reply pkg/kopia/repository/client.go:66]: @ankitjain235 In [this](https://github.com/kopia/kopia/pull/3238) Kopia PR for adding hard limits for cache, the soft limit fields in the struct were renamed from `MaxCacheSizeBytes` to `ContentCacheSizeBytes` and `MaxMetadataCacheSizeBytes` to `MetadataCacheSizeBytes`. Since Le's changes are only for kopia upgrade, we are using renamed fields for Soft limits. The actual work of replacing soft limits with hard limits fields would be done in current PR. I wanted to keep the changes of upgrade separate from the changes that are required for hard limit flags. 

So Le's PR should be merged first and then current PR would be merged

**Review by ankitjain235 — APPROVED** [review]
LGTM, should be good to be merged once testing is done.
Approving to unblock since I had requested changes.

File: pkg/kopia/command/cache.go
Comment (kale-amruta) [reply pkg/kopia/command/cache.go:30]: we are using `c.ContentCacheLimitMB` and `c.MetadataCacheLimitMB` which is a type of cacheArgs

File: pkg/kopia/command/cache.go
Comment (pavannd1) [inline pkg/kopia/command/cache.go:35]:
```suggestion
	// The hard limit flags for cache would be set using the env variables that
	// are passed through Kanister helm settings.
	// Soft limit flags would be set to the default values by kopia
	// automatically in ``connectOptions.setup()`` function.
	// Refer to https://github.com/kopia/kopia/blob/master/cli/command_repository_connect.go#L71
```

File: pkg/kopia/command/repository_test.go
Comment (pavannd1) [reply pkg/kopia/command/repository_test.go:49]: I guess we are trying to keep everything the same as before. We can revisit this later too.

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**
